### PR TITLE
Issue #4 - added default configuration fetching API

### PR DESCRIPTION
**Pull request by magsout**
@miketaylr I starting adding the functions/configurations fetching the API.

I think using redux could be a good thing.

I have in my mind this:
- Components + containers -> https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0
- action + reducers: handle data with a logical Entity and send datas into containers

What do you think ?

### CODE REVIEW

File: client/src/api/request/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  }
+  if ("function" != typeof callback) {
+    throw new Error(`callback - ${callback} - is not a function`);
+  }
+  return dispatch => {
+    return jsonFetch(requestUrl, requestParameters).then(
+      response => {
+        dispatch(callback(response, requestParameters));
+      },
+      response => {
```
Comment (miketaylr) [inline client/src/api/request/index.js]: (im obviously not familiar with this module... can you explain why we dispatch the callback twice?)

File: client/src/api/request/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        dispatch(callback(response, requestParameters, true));
```
Comment (miketaylr) [inline client/src/api/request/index.js]: Ah, this is the error callback. I guess we're using this `true` to mean it's an error or something?

Comment (magsout) [reply client/src/api/request/index.js]: yes exactly, `jsonFetch` returns a `Promise((resolve, reject)`. So I pass a boolean in my callback. false: no error,  true: errors

**Comment by magsout** [comment]
thanks @miketaylr  <3

**Pull request merged**